doc/webdav-intro.md

```diff
@@ -0,0 +1,281 @@
+  user.
+
+- An "href" element inside a "response" contains the URL of the resource that
+  the response describes, as either a complete URL or as the absolute path
+  component of a URL.  In the latter case, the path must begin with a forward
+  slash and include all path components "leading up to" the resource from the
+  server root, and the complete URL is calculated by resolving this path
+  against the URL of the request (roughly, by combining the non-path portions
+  of the request URL with the path).  All "href" elements values within a
+  Multi-Status response must be unique and must use the same choice of URL vs.
+  absolute path.
+
+- The "propstat" elements inside a "response" contain the names and possibly
+  also values of the resource's properties, along with a status indicating the
+  success of the request for the information.  Each "propstat" groups together
+  properties that share the same status (though the RFC does not seem to
+  require that the groups be maximal).
+
+    - A "prop" element contains one or more property elements (see "Properties"
+      below).  If the request was not for "propname" and the server was able to
+      successfully determine the values of the given properties, the elements
+      will contain the corresponding properties' values; otherwise, the
+      elements will be empty.
+
+    - A "status" element contains an HTTP status line (e.g., "`HTTP/1.1 200
+      OK`" or "`HTTP/1.1 404 NOT FOUND`") describing the server's success in
+      obtaining the properties in the sibling "prop" element.  The status code
+      may be any code in the 2xx, 3xx, 4xx, or 5xx range (though it is unclear
+      when a 3xx code would be used).
+
+- It appears that the `(href*, status)` production listed as an alternative to
+  "propstat" in the DTD is only used for responses to verbs other than
+  `PROPFIND`; the RFC isn't 100% clear.
+
+- The "location" element is seemingly only used in conjunction with 3xx
+  "status" elements immediately inside a response, and thus they presumably do
+  not apply to `PROPFIND` responses; the RFC isn't clear.
+
+- "error" elements contain precondition or postcondition codes identifying a
+  violated requirement.  None of the codes defined in RFC 4918 would appear in
+  an "error" inside a Multi-Status response to a `PROPFIND` request.
+
+See [RFC 4918, §9.1.3 *et sequentes*][examples] for some examples of `PROPFIND`
+response bodies.
+
+
+The `Dav` Header
+----------------
+
+A response to an `OPTIONS` request to a WebDAV-enabled URL must include a "Dav"
+header whose value is a comma-separated list of tokens identifying the
+"compliance classes" that the resource supports.  Class "1" is for resources
+that meet all "MUST" requirements in the RFC.  Class "2" is for resources that
+support class "1" and also various locking-related functionality.  Class "3" is
+for resources that support class "1" and also the changes made to RFC 2518 by
+RFC 4918.
```

Comment:
Is there some WebDAV validator/tester , to check WebDAV resource on compliance? Not unlike 'git annex testremote' testing compliance to annex external protocol?

Reply:
I'm not aware of one, and I don't see one on http://webdav.org/projects/.